RecoMET/METProducers/src/HcalNoiseInfoProducer.cc

```diff
@@ -500,9 +707,134 @@ HcalNoiseInfoProducer::filldigis(edm::Event& iEvent, const edm::EventSetup& iSet
 	      }
           } // end of HBHE check
      } // loop on HcalCalibDigiCollection
+
+     // now match the laser monitor data by fiber (in time) 
+     if( fillLaserMonitor_ ) {
+       // check for any fibers without data and fill
+       // them so we dont run into problems later
+       for( unsigned idx = 0; idx < laserMonCBoxList_.size(); ++idx ) {
+           if( lasmon_adcs[idx].empty() ) {
+               lasmon_adcs[idx] = std::vector<int>(10, -1);
+           }
+           if( lasmon_capids[idx].empty() ) {
+               lasmon_capids[idx] = std::vector<int>(10, -1);
+           }
+       }
+       unsigned nFibers = laserMonIEtaList_.size();
+       // for each fiber we need to find the index at with the 
+       // data from the next fiber matches in order to stitch them together.
+       // When there is an overlap, the data from the end of the
+       // earlier fiber is removed.  There is no removal of the last fiber
+       std::vector<unsigned> matching_idx; 
+       // we assume that the list of fibers was given in time order
+       // (if this was not the case, then we just end up using 
+       // all data from all fibers )
+       for( unsigned fidx = 0; fidx < (nFibers - 1); ++fidx ) {
+
+         // start with the last TS and loop backwards
+         // on each iteration check if all capId and ADCs from this
+         // TS forward match the beginning entries of 
+         // the next fiber
+         unsigned last_ts = lasmon_capids[fidx].size()-1;  // last TS
+         int start_ts = last_ts; // start_ts will be decrimented on each loop
+         // in the case that our stringent check below doesn't work 
+         // store the latest capID that has a match
+         int latest_cap_match = -1;
+         // reverse loop over TSs
+         for( unsigned ncheck = 1; ncheck <= lasmon_capids[fidx].size() ; ncheck++ ) {
+           unsigned nmatch_cap = 0; // count the number of TS where capID matched
```

Comment:
Probably, the whole block between L745 and L767 could be simplified as follow (without changing the logic):
```
           // loop over the channel TS, this is for the later fiber in time
           bool cap_match = true;
           bool adc_match = true;
           for( unsigned lidx = 0; lidx < ncheck; lidx++) {
             // we are looping over the TS of the later fiber in time
             // the TS of the earlier fiber starts from the end
             unsigned eidx = last_ts-ncheck+lidx;
             // if we get an invald value, this fiber has no data
             // the check and match will fail, so the start_ts will 
             // be decrimented
             if( lasmon_capids[fidx][eidx] == -1 || lasmon_capids[fidx+1][lidx] == -1 ) {
               cap_match = false;
               adc_match = false;
	       break;
	     }
             if( lasmon_capids[fidx][eidx] == lasmon_capids[fidx+1][lidx] ) {
	       cap_match = false;
             }
             // check the data values as well
             if( lasmon_adcs[fidx][eidx] == lasmon_adcs[fidx+1][lidx] ) {
	       adc_match = false;
             }
           }
```

Reply:
I think it is possible.  Whenever a match fails the bool is set to false.  I think in the code above you mean to have != instead of == 

             if( lasmon_capids[fidx][eidx] != lasmon_capids[fidx+1][lidx] ) {
	       cap_match = false;
             }
             // check the data values as well
             if( lasmon_adcs[fidx][eidx] != lasmon_adcs[fidx+1][lidx] ) {
	       adc_match = false;
             }

Reply:
Yes, indeed...
Thank you for noticing